src/idl_parser.cpp

```diff
@@ -28,6 +28,79 @@
 #include "flatbuffers/idl.h"
 #include "flatbuffers/util.h"
 
+#if defined(FLATBUFFERS_CPP98_STL)
+#include <cctype>
+#endif  // defined(FLATBUFFERS_CPP98_STL)
+
+// Some C++98 implementations (e.g stlport) do not define
+// std::numeric_limits for 64-bit types.
+namespace flatbuffers {
+#ifndef FLATBUFFERS_CPP98_STL
+#if !(defined(_MSC_VER) && _MSC_VER <= 1700 /* MSVC2012 */)
+template <typename T>
+using numeric_limits = std::numeric_limits<T>;
+#else
+template <typename T>
+class numeric_limits : public std::numeric_limits<T> {};
+#endif  // !(defined(_MSC_VER) && _MSC_VER <= 1700 /* MSVC2012 */)
+#else
+template <typename T>
+class numeric_limits : public std::numeric_limits<T> {};
+
+template <>
+class numeric_limits<unsigned long long> {
+ public:
+  static unsigned long long min() { return 0ULL; }
+  static unsigned long long max() { return ~0ULL; }
+};
+
+template <>
+class numeric_limits<long long> {
+ public:
+  static long long min() {
+    return static_cast<long long>(1ULL << ((sizeof(long long) << 3) - 1));
+  }
+  static long long max() {
+    return static_cast<long long>((1ULL << ((sizeof(long long) << 3) - 1)) - 1);
+  }
+};
+#endif  // FLATBUFFERS_CPP98_STL
+
+// MSVC 2010 does not allow lambdas (with no captures) to be passed as plain
+// C++ functions so instatitate a wrapper that forwards to the lambda.
+template<typename F>
+class ParseDelimitersCallbackHolder {
```

Comment:
Unfortunately it can't be a macro as a lambda in <= MSVC2012 can't just be cast to a function pointer it seems to be a pretty broken state of affairs.  I could split the logic out into static methods rather than using lambdas but it makes the code harder to follow WDYT?